modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_8_0/NotificationQueueEntryUpgradeProcess.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.notification.internal.upgrade.v3_8_0;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Selton Guedes
+ */
+public class NotificationQueueEntryUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				StringBundler.concat(
+					"select distinct NotificationQueueEntry.",
```

Comment:
Why did you need to add the **distinct** in the query?

Reply:
In this query we need `notificationTemplateId` and `companyId` to be used to populate the companyId of the NotificationQueueEntry.
We are doing a select in the NotificationQueueEntry table, so we will return all the `notificationTemplateId` and `companyId` of the NotificationQueueEntry instances. But in this case we can have more than one NotificationQueueEntry instaces with the same `notificationTemplateId` and `companyId`. I'm using the distinct to avoid returning duplicate instances in this case.